Dev/kernel functions docs update (#2312)
oneapi-src/oneDAL: docs/source/onedal/algorithms/kernel-functions/index.rst

Kernel Functions
================

> A kernel function is function that takes input vectors from original space
> and returns dot product of the vectors in the feature space.
> More formally, if we have :math:`x,y \in X`, and :math:`\phi \in X \leftarrow \mathbb{R}^n`,
> then kernel function

Reviewer: ```suggestion
A kernel function is function that takes input vectors from original :math:`p`-dimensional space
and returns dot product of the vectors in :math:`P`-dimensional feature space.
More formally, if we have :math:`x,y \in \mathbb{R}^p`, and :math:`\phi \in \mathbb{R}^p \leftarrow \mathbb{R}^P`,
then kernel function
```

Kernel Functions
================

A kernel function takes input vectors from original (:math:`p`-dimensional) space

> and returns dot product of the vectors in :math:`s`-dimensional feature space.

Reviewer 2: ```suggestion
and returns the dot product of the vectors in the :math:`s`-dimensional feature space.
```

> More formally, if we have :math:`x,y \in \mathbb{R}^p`, and

Reviewer 2: ```suggestion
For example, if you have :math:`x,y \in \mathbb{R}^p`, and
```

Author: This is not an example, but a mathematical definition, which was described with words above. 
So changed to
`Thus, having :math:....`

> :math:`\phi \in \mathbb{R}^p \leftarrow \mathbb{R}^s`, then kernel function is

Reviewer 2: ```suggestion
:math:`\phi \in \mathbb{R}^p \leftarrow \mathbb{R}^s`, then the kernel function is
```

.. math::
   K(x, y) \equiv <\phi (x), \phi (y)>
.. _kernel_func_def:

> where :math:`\phi(x)` is a vector-valued function:

Reviewer 2: ```suggestion
Where :math:`\phi(x)` is a vector-valued function:
```

:math:`\phi(x) = \phi_1(x), \phi_2(x), \ldots, \phi_s(x)`.

> In case, when :math:`\phi (x) = x`, the kernel is linear.

Reviewer 2: ```suggestion
If :math:`\phi (x) = x`, the kernel is linear.
```

> Kernels are used in SVM model, but for some tasks it could be used separately to transform vectors

Reviewer 2: ```suggestion
Kernels are used in the SVM model, but for some tasks, they could be used separately to transform vectors
```
